app/models/resource_action_workflow.rb

```diff
@@ -59,7 +59,7 @@ def validate_dialog
 
   def create_values
     create_values_hash.tap do |value|
-      value[:src_id] = @target.id
+      value[:src_id] = @target.try(:id)
```

Comment:
@smelamud 
Since you are doing bulk migration wouldn't you still want to monitor the workflow of one item at a time. The workflow should be fed 1 item at a time with the target set to the VM being migrated. 
If you have blocked @target here but there could be other places where the @target would be important.
https://github.com/ManageIQ/manageiq/blob/master/app/models/resource_action_workflow.rb#L46
When we do custom actions where a user selects a group of vms (similar to bulk migration) we feed the objects 1 at a time e.g. here
https://github.com/ManageIQ/manageiq/blob/master/app/models/resource_action.rb#L23
Which also allows us to do them 1 by 1 or pass all the IDs into Automate and it can do them 1 at a time.
Doing them 1 at a time allows for individual tracking.

Reply:
@smelamud 
You are iterating thru the VM's here in the Automate Method
https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/create_vm_import_request.rb#L22
So does this call happen the first time.

Reply:
@smelamud 
From the log attached to the BZ you have this line which is also suspicious

```
 MIQ(MiqAeEngine.deliver) Delivering {"dialog_tag_category"=>"", "dialog_tag_name"=>"", "dialog_name"=>"", "dialog_provider"=>"3", "dialog_cluster"=>"4", "dialog_storage"=>"22", "dialog_sparse"=>"f", "dialog_install_drivers"=>"f", "dialog_drivers_iso"=>""} for object [.] with state [] to Automate
```
Usually when we can Automate we pass in a VMDB object which I am guessing is the target and the @target is nil in your case.

A dialog needs to be anchored to some VMDB object (host, vm , storage, ems, ...) In your case I think the dialog is not attached to anything leading to target being nil.

@tinaafitz How does the initial launch of this process compare to the migration when we have a single vm. Should this bulk migration dialog be anchored to an EMS. so the target would be the EMS we are working with

Reply:
@smelamud 
When @matobet was implementing this for a single VM he had the target set to the selected vm
https://github.com/ManageIQ/manageiq-ui-classic/pull/383/files
In your case you are doing a bulk VM migration you would set your target to be one of the source EMS

In here https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/actions/vm_actions/transform.rb#L17
When you are doing bulk migration pass in the EMS object as the target.
For a single VM migration you pass the target as VM
for bulk VM migration you pass the target as the EMS